src/ui/server/controllers/insurance/business/company-details/index.ts

```diff
@@ -139,18 +147,32 @@ const post = async (req: Request, res: Response) => {
 
     /**
      * If "different trading address" has been submitted as "yes"/true,
-     * Redirect to the "alternative trading address" route.
+     * Redirect to the "alternative trading address" route if not a check/check or change route.
      */
-    if (submittedValues[TRADING_ADDRESS]) {
+    if (submittedValues[TRADING_ADDRESS] && !(isChangeRoute(req.originalUrl) || isCheckAndChangeRoute(req.originalUrl))) {
       return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${ALTERNATIVE_TRADING_ADDRESS_ROOT}`);
     }
 
+    // check and check-and-change routes
+    let changeRoute = `${INSURANCE_ROOT}/${referenceNumber}${CHECK_YOUR_ANSWERS}`;
+    let checkAndChangeRoute = `${INSURANCE_ROOT}/${referenceNumber}${CHECK_AND_CHANGE_ROUTE}`;
+
+    // if is a change route and TRADING_ADDRESS is true, then redirect ALTERNATIVE_TRADING_ADDRESS_CHANGE
+    if (submittedValues[TRADING_ADDRESS] && isChangeRoute(req.originalUrl)) {
+      changeRoute = `${INSURANCE_ROOT}/${referenceNumber}${ALTERNATIVE_TRADING_ADDRESS_CHANGE}`;
+    }
+
+    // if is a check-and-change route and TRADING_ADDRESS is true, then redirect ALTERNATIVE_TRADING_ADDRESS_CHECK_AND_CHANGE
+    if (submittedValues[TRADING_ADDRESS] && isCheckAndChangeRoute(req.originalUrl)) {
+      checkAndChangeRoute = `${INSURANCE_ROOT}/${referenceNumber}${ALTERNATIVE_TRADING_ADDRESS_CHECK_AND_CHANGE}`;
+    }
+
```

Comment:
Can we update these comments to be multi line? easier to read/digest so its "if ... then", e.g

```js

/**
 * If is a change route and TRADING_ADDRESS is true,
 * redirect ALTERNATIVE_TRADING_ADDRESS_CHANGE
 */
```